crypto/rsa/rsa_ameth.c

```diff
@@ -60,13 +60,16 @@ static int rsa_pub_encode(X509_PUBKEY *pk, const EVP_PKEY *pkey)
     if (!rsa_param_encode(pkey, &str, &strtype))
         return 0;
     penclen = i2d_RSAPublicKey(pkey->pkey.rsa, &penc);
-    if (penclen <= 0)
+    if (penclen <= 0) {
+        ASN1_STRING_free(str);
```

Comment:
```
if (penclen < 0)
    goto out
...
out:
ASN1_STRING_free(str)
return 0
```
instead maybe?
of alternatively
```
if (penclen > 0) {
     if (X509_PUBKEY_set0_param(...))
         return 1;
}
ASN1_STRING_free(str);
OPENSSL_free(penc);
return 0;
```

Just to avoid having two locations where we have to free the string?